tests/install_testbed/launch_vm.py

```diff
@@ -0,0 +1,192 @@
+    with the default neutron configuration).
+
+    Be aware that this port is not deleted when the VM is deleted!
+
+    Another point is that security group may be different than the security
+    group of the VM.
+
+    :param security_group_id: The security group
+    :return: a pre-allocated port
+    """
+
+    subnet_external = None
+    for subnet in neutron.list_subnets()['subnets']:
+        if subnet['network_id'] == network['external']:
+            subnet_external = subnet['id']
+
+    fixed_ips = list()
+    for i in range(200, 205):
+        fixed_ip = dict()
+        fixed_ip['subnet_id'] = subnet_external
+        fixed_ip['ip_address'] = '192.168.58.%d' % i
+        fixed_ips.append(fixed_ip)
+
+    p = {'port': {'network_id': network['external'], 'fixed_ips': fixed_ips}}
+    if security_group_id:
+        p['port']['security_groups'] = [security_group_id]
+    return neutron.create_port(p)
+
+# Get networks
+
+network = dict()
+neutron = osclients.get_neutronclient()
+nova = osclients.get_novaclient()
+
+for net in neutron.list_networks()['networks']:
+    for n in settings.network_names:
+        if net['name'] == settings.network_names[n]:
+            network[n] = net['id']
+
+for n in settings.network_names:
+    if not n in network:
+        if n == 'management':
+            sys.stderr.write('Fatal error: network ' + settings.network_names[n] +
+                             'not found.\n')
+            sys.exit(-1)
+
+        network[n] = neutron.create_network(
+            {'network': {'name': settings.network_names[n], 'admin_state_up': True}})['network']['id']
+        # create subnetwork. It is not possible to assign a network
+        # to a VM without a subnetwork.
+        neutron.create_subnet({'subnet': {'network_id': network[n],
+            'ip_version': 4, 'cidr': settings.subnet[n], 'gateway_ip': None}})
+
+# Get a floating IP
+floating_ip = None
+for ip in neutron.list_floatingips()['floatingips']:
+    floating_ip = ip['floating_ip_address']
+    if not settings.preferred_ip or settings.preferred_ip == floating_ip:
+        break
+
+if not floating_ip:
+    # allocate floating ip if it does not exist
+    floating_ip = nova.floating_ips.create('ext-net').ip
+
+keys = nova.keypairs.findall(name=settings.key_name)
+if not keys:
+    new_key = nova.keypairs.create(settings.key_name)
+    filename = os.path.expanduser('~/.ssh/' + settings.key_name)
+    with open(filename, 'w') as f:
+        f.write(new_key.private_key)
+    os.chmod(filename, 0600)
+
+# Create security group if it does not exist
+sg_name = settings.security_group
+sec_groups = nova.security_groups.findall(name=sg_name)
+if not sec_groups:
+    g = nova.security_groups.create(name=sg_name, description=sg_name)
+    nova.security_group_rules.create(
+        g.id, ip_protocol='icmp', from_port=-1, to_port=-1, cidr='0.0.0.0/0')
+    nova.security_group_rules.create(
+        g.id, ip_protocol='tcp', from_port=22, to_port=22, cidr='0.0.0.0/0')
```

Comment:
I'll put a comment. I think is more clear than defining a constant on the top of the file. Isn't it?